source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename)
+{
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D setColorInObjectModel3D(HalconCpp::HImage RGB, HalconCpp::HObjectModel3D pcHalcon) {
+
+	HalconCpp::HImage R = RGB.AccessChannel(1);
+	HalconCpp::HImage G = RGB.AccessChannel(2);
+	HalconCpp::HImage B = RGB.AccessChannel(3);
+
+	HalconCpp::HRegion domain = R.GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	HalconCpp::HTuple Rtup = R.GetGrayval(rows, cols);
+	HalconCpp::HTuple Gtup = G.GetGrayval(rows, cols);
+	HalconCpp::HTuple Btup = B.GetGrayval(rows, cols);
+
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+	return pcHalcon;
+}
+
+
+int main()
+{
+	try
+	{
+		std::cout << "Connecting to camera" << std::endl;
+
+		// Connect to camera using GeniCamTL driver
+		HalconCpp::HFramegrabber Framegrabber;
+		Framegrabber.OpenFramegrabber("GenICamTL", 1, 1, 0, 0, 0, 0, "progressive", -1, "default", -1, "false", "default", "default", 0, 0);
```

Comment:
First, implement the following:
https://github.com/zivid/halcon-samples/blob/362cae911796c211ead96ea469c33d43b66a0617/source/Camera/Basic/Capture.hdev#L25-L28
https://github.com/zivid/halcon-samples/blob/master/source/Procedures/get_first_available_zivid_device.hdvp
get the device and then use it in the OpenFramegrabber. This will prevent from connecting to the ToshibaTeli directly.

Reply:
Implemented this as a function. It was not trivial to understand how Hstring and HTuple works, but I am able to connect to Zivid camera, so I think it works